selfie.c

```diff
@@ -2286,25 +2288,32 @@ uint64_t isCharacterWhitespace() {
     return 1;
   else if (character == CHAR_TAB)
     return 1;
+  else if (character == CHAR_EOF)
+    return 1;
   else
     return isCharacterNewLine();
 }
 
 uint64_t findNextCharacter() {
-  uint64_t inComment;
+  uint64_t inSingleLineComment;
+  uint64_t inMultiLineComment;
+  uint64_t readWhitespaceInMultiLineComment;
```

Comment:
Can you avoid using such a flag? How about first checking `inSingleLineComment`, then `inMultiLineComment`, then whitespace, then slash?

Reply:
That would require to check for line feeds and EOF in `inMultiLineComment` again, since both read characters in `inMultiLineComment` could be those two and `isCharacterWhitespace()` is never reached. And in turn the code duplication would be back.

Reply:
Interesting. I am wondering if there is an abstraction that would simplify the code and avoid code duplication.

Reply:
The `lineNumber` could be incremented in `getCharacter()`, that would mean no increments anywhere else and would only leave EOF.